.github/workflows/pr-check-build-che-code-image.yaml

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) 2023 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
+name: Build che-code images Pull Request check
+
+# Trigger the workflow on pull request
+on:
+  pull_request:
+    paths-ignore:
+      - '**/*.md'
+      - devfile.yaml
+      - '.che/*'
+      - '.rebase/*'
+
+jobs:
+  build-che-code:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Checkout che-code source code
+        uses: actions/checkout@v2
+
+      - name: Login to Quay.io
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_PULL_REQUESTS_USERNAME }}
+          password: ${{ secrets.QUAY_PULL_REQUESTS_PASSWORD }}
+
+      - name: Compile linux-libc
```

Comment:
If we build libc and musl images [in parallel](https://github.com/che-incubator/che-code/blob/3aa3c9d2dc54ce332121fef82c72cf530fe8ca1e/.github/workflows/pr-check.yml#L22) the whole workflow will take about half as long to complete.

https://github.com/che-incubator/che-code/actions/runs/6111546375
This ^^ should take ~1 hour instead of ~2

Reply:
> It could happen I want to rebuild only libc or musl.

agree, and it's currently possible
because [the matrix strategy splits the `build` job](https://github.com/che-incubator/che-code/blob/dfeffd73c46760906aca16e25f306a33f4cdbb2e/.github/workflows/pr-check.yml#L17-L23) and makes it possible to have separate jobs for libc and musl variants:

<img width="336" alt="image" src="https://github.com/che-incubator/che-code/assets/1636395/c5a98ce8-7b86-4926-807f-ca7fb5107c69">

But with the changes from this PR, we lose such a possibility because everything will run as a single job:

<img width="336" alt="image" src="https://github.com/che-incubator/che-code/assets/1636395/f5ac0cd2-f323-439a-a5de-e4e044c204ac">

For the developer, it means that in case of any build failures whole workflow must be rerun from the beginning and the dev will need to wait for a couple of more hours.

Reply:
I was confused a bit and thought we were discussing devfile commands, will try to rework this action.

Reply:
If we use matrix build, the intermediate result will be stored to artifacts like here
https://github.com/che-incubator/che-code/blob/main/.github/workflows/pr-check.yml#L38-L41

As I remember we discussed that and decided not to use artifacts to simplify the process.
Would you like to revert it back?

Reply:
Considering that storing intermediate containers as GH artifacts gives us such a nice dev UX, I'd keep using artifacts. By "nice dev UX" I mean:
- having parallel jobs with the possibility of rerunning just one of them in case of any build failure
- keeping the whole GH Workflow at least x2 faster

Reply:
Cool! Thanks @vitaliy-guliy 👍